src/python/pants/engine/README.md

```diff
@@ -17,63 +17,69 @@ Once the engine is instantiated with a valid set of `@rule`s, a caller can synch
 computation of any of the product types provided by those `@rule`s by calling:
 
 ```python
-# Request a ThingINeed (a `Product`) for the thing_i_have (a `Subject`).
+# Request a ThingINeed (a `Product`) for a thing_i_have (a `Param`).
 thing_i_need, = scheduler.product_request(ThingINeed, [thing_i_have])
```

Comment:
No: it's unpacking a single result from a list. I feel like I like this syntax better than:
```
thing_i_need = scheduler.product_request(ThingINeed, [thing_i_have])[0]
```
... but if we want to avoid this, we can.

Reply:
`thing_i_need = scheduler.product_request(ThingINeed, [thing_i_have])[0]` is less surprising to me. I recommend using that style both in documentation and source code.

Reply:
The comma is subtle.Perhaps `.pop()` as the idiom is a compromise, but it would seem nicer to assert a single item which the tuple unpack does. Maybe wrapping this case up in an API pulls its weight and `scheduler.single_product_request(ThingINeed, [thing_i_have])` should be a thing (I think it was at some point?). Maybe `request` and `request_single` would be better names - the product bit is perhaps redundant - the only thing you can request is a product.